packages/react/src/analytics/integrations/Forter/__tests__/Forter.test.ts

```diff
@@ -0,0 +1,209 @@
+        loadIntegrationData,
+        strippedDownAnalytics,
+      ),
+    ).toBeInstanceOf(Forter);
+  });
+
+  it('should return an error when siteId is not set on Forter Options.', () => {
+    // @ts-expect-error try loading forter without passing siteId
+    Forter.createInstance({}, loadIntegrationData, strippedDownAnalytics);
+
+    expect(utils.logger.warn).toHaveBeenCalledWith(
+      "[Forter] - `siteId` parameter was not provided in options. It's mandatory to load Forter Integration.",
+    );
+  });
+
+  it('should log a warn message if `origin` is not specified in options', () => {
+    Forter.createInstance(
+      { siteId: mockSiteId },
+      loadIntegrationData,
+      strippedDownAnalytics,
+    ) as Forter;
+
+    expect(utils.logger.warn).toHaveBeenCalledWith(
+      "[Forter] - `origin` parameter was not provided in options. It's advisable to provide an origin option to aid in debugging",
+    );
+  });
+
+  it('should post a message to Omnitracking with forter token when it is loaded', async () => {
+    const createEventMock = async function (
+      type: TrackTypesValues,
+      event: string,
+    ) {
+      return Promise.resolve({
+        ...eventData,
+        event,
+        type,
+      });
+    };
+
+    const forterInstance = Forter.createInstance(
+      {
+        siteId: mockSiteId,
+        origin: mockOrigin,
+      },
+      loadIntegrationData,
+      { ...strippedDownAnalytics, createEvent: createEventMock },
+    );
+    // @ts-expect-error accessing private property just for testing
+    const spy = jest.spyOn(forterInstance, 'resolveOnSetUserPromise');
+    // @ts-expect-error test data
+    const data: SetUserEventData = {};
+
+    forterInstance.onSetUser(data);
+
+    await mockPostTrackingsPromise;
+
+    expect(postTracking).toHaveBeenCalledWith(
+      expect.objectContaining({
+        tenantId: eventData.context.tenantId,
+        clientId: eventData.context.clientId,
+        correlationId: eventData.user.localId,
+        customerId: `${eventData.user.id}`,
+        event: 'PageAction',
+        parameters: expect.objectContaining({
+          clientTimestamp: expect.any(String),
+          tid: ForterTokenTid,
+          uuid: expect.any(String),
+          val: JSON.stringify({
+            forterTokenCookie: mockForterToken,
+            origin: mockOrigin,
+            // @ts-expect-error
+            userAgent: eventData?.context?.web?.window?.navigator?.userAgent,
+          }),
+        }),
+      }),
+    );
+
+    jest.clearAllMocks();
+
+    forterInstance.onSetUser(data);
+
+    expect(spy).toHaveBeenCalledTimes(0);
+  });
+
+  it('should post a message to Omnitracking with forter using custom http client', async () => {
+    const createEventMock = async function (
+      type: TrackTypesValues,
+      event: string,
+    ) {
+      return Promise.resolve({
+        ...eventData,
+        event,
+        type,
+      });
+    };
+
+    jest.clearAllMocks();
```

Comment:
This is unnecessary since there is a `beforeEach` block that will call `jest.clearAllMocks` already.